controllers/clustergroupupgrade_controller.go

```diff
@@ -361,43 +341,87 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 			nextReconcile = requeueWithMediumInterval()
 		} else {
 			// There are no blocking CRs, continue with the upgrade process.
-			// Take actions before starting upgrade.
-			err = r.takeActionsBeforeEnable(ctx, clusterGroupUpgrade)
+			// create backup
+			err = r.reconcileBackup(ctx, clusterGroupUpgrade)
 			if err != nil {
+				r.Log.Error(err, "reconcileBackup error")
 				return
 			}
 
-			// If the remediation plan is empty, update the status.
-			if clusterGroupUpgrade.Status.RemediationPlan == nil {
-				utils.SetStatusCondition(
-					&clusterGroupUpgrade.Status.Conditions,
-					utils.ConditionTypes.Progressing,
-					utils.ConditionReasons.Completed,
-					metav1.ConditionFalse,
-					"The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
-				)
-				utils.SetStatusCondition(
-					&clusterGroupUpgrade.Status.Conditions,
-					utils.ConditionTypes.Succeeded,
-					utils.ConditionReasons.UpgradeCompleted,
-					metav1.ConditionTrue,
-					"The ClusterGroupUpgrade CR has all clusters already compliant with the specified managed policies",
-				)
-				nextReconcile = requeueImmediately()
-			} else {
-				// Start the upgrade.
-				utils.SetStatusCondition(
-					&clusterGroupUpgrade.Status.Conditions,
-					utils.ConditionTypes.Progressing,
-					utils.ConditionReasons.InProgress,
-					metav1.ConditionTrue,
-					"The ClusterGroupUpgrade CR has upgrade policies that are still non compliant",
-				)
-				clusterGroupUpgrade.Status.Status.StartedAt = metav1.Now()
-				nextReconcile = requeueImmediately()
+			if clusterGroupUpgrade.Status.Backup != nil {
+				for _, v := range clusterGroupUpgrade.Status.Backup.Status {
+					//nolint
+					if v == BackupStatePreparingToStart || v == BackupStateStarting || v == BackupStateActive {
+						utils.SetStatusCondition(
+							&clusterGroupUpgrade.Status.Conditions,
+							utils.ConditionTypes.Progressing,
+							utils.ConditionReasons.NotStarted,
+							metav1.ConditionFalse,
+							"The Cluster backup is in progress",
+						)
+						err = r.updateStatus(ctx, clusterGroupUpgrade)
+						nextReconcile = requeueWithShortInterval()
+						return
+					}
+				}
 			}
-		}
 
+			backupCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.BackupSuceeded))
+			if clusterGroupUpgrade.Status.Backup == nil || (backupCondition != nil && backupCondition.Status == metav1.ConditionTrue) {
```

Comment:
- **when backup is running it will be false:**
    **- false || (true && false)**
- when backup is false it will be true:
    - true || (false && false)
- when backup is done it will be true:
    - true || (true && true)

Reply:
when backup is running, don't we always return from line 364?

Reply:
Yeah, that is on purpose, when backup is progressing nothing else should happen in the SNO, we should always return from that point. When backup is done, failed or in error, only then we move to next block. This part remained the same as it was